relation_engine/taxa/ncbi/test/parsers_test.py

```diff
@@ -168,13 +178,247 @@ def test_node_provider_complex():
             'scientific_name':            "sciname74",
             'rank':                       "no rank",
             'strain':                     False,
+            'species_or_below':           False,
             'aliases':                    [],
             'ncbi_taxon_id':              74,
             'gencode':                    11,
         },
     ]
 
 
+_NON_SPECIES_RANKS = [
+    "superkingdom",
+    "kingdom",
+    "subkingdom",
+    "superphylum",
+    "phylum",
+    "subphylum",
+    "infraphylum",
+    "superclass",
+    "class",
+    "subclass",
+    "infraclass",
+    "cohort",
+    "subcohort",
+    "superorder",
+    "order",
+    "suborder",
+    "infraorder",
+    "parvorder",
+    "superfamily",
+    "family",
+    "subfamily",
+    "tribe",
+    "subtribe",
+    "genus",
+    "subgenus",
+    "section",
+    "subsection",
+    "series",
+    "subseries",
+    "species group",
+    "species subgroup",
+]
+
+
+def test_node_provider_non_species_ranks():
+    # test that non-species ranks are accepted and don't result in marking non hierarchical ranks
+    # below them as strains.
+    # This test duplicates the list of ranks from ranks.py, which it should - if someone screws
+    # up ranks.py this test should fail. It will not if it just imports and uses the same
+    # rank set.
+    for rank in _NON_SPECIES_RANKS:
+        names = StringIO("\n".join([
+            "  \t 5  |  \t  name5   |    |   scientific name   \t  | ",
+            "  \t 10  |  \t  name10   |    |   scientific name   \t  | ",
+            "  \t 15  |  \t  name15   |    |   scientific name   \t  | ",
+            "  \t 100  |  \t  name100   |    |   scientific name   \t  | ",
+            "  \t 105  |  \t  name105   |    |   scientific name   \t  | ",
+            "  \t 110  |  \t  name110   |    |   scientific name   \t  | ",
+        ]))
+        nodes = StringIO("\n".join([
+            f" 5	|	1	|	{rank}   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
+            " 10	|	5	|	clade   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
+            " 15	|	10	|	no rank   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
+            " 100	|	678	|	varietas   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
+            " 105	|	100	|	no rank   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
+            " 110	|	105	|	clade   	|		|	8	|	0	|	1	|	0	|	0	|	0	|	0	|0	|		|",
```

Comment:
Maybe it's just github but these string tables render quite illegibly due to line wrapping, whitespace, etc

Reply:
maybe triple-quote strings with textwrap.dedent, or 2d arrays with a helper function to join them?

Reply:
These are copied directly from the NCBI input files, and I wanted to leave them as is because of the bizarro whitespace / separators they use - make sure the tests pass on that stuff